x/ccv/provider/keeper/hooks_test.go

```diff
@@ -76,3 +89,137 @@ func TestValidatorConsensusKeyInUse(t *testing.T) {
 		})
 	}
 }
+
+func TestAfterPropSubmissionAndVotingPeriodEnded(t *testing.T) {
+	acct := cryptotestutil.NewCryptoIdentityFromIntSeed(0)
+
+	propMsg, err := v1.NewLegacyContent(
+		testkeeper.GetTestConsumerAdditionProp(),
+		authtypes.NewModuleAddress("gov").String(),
+	)
+	require.NoError(t, err)
+
+	prop, err := v1.NewProposal(
+		[]sdk.Msg{propMsg},
+		0,
+		time.Now(),
+		time.Now(),
+		"",
+		"",
+		"",
+		sdk.AccAddress(acct.SDKValOpAddress()),
+	)
+	require.NoError(t, err)
+
+	k, ctx, ctrl, mocks := testkeeper.GetProviderKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
+	defer ctrl.Finish()
+
+	// pass the prop to the mocked gov keeper,
+	// which is called by both the AfterProposalVotingPeriodEnded and
+	// AfterProposalSubmission gov hooks
+	gomock.InOrder(
+		mocks.MockGovKeeper.EXPECT().GetProposal(ctx, prop.Id).Return(prop, true).Times(2),
```

Comment:
2 times because first time it's executed in a cached context and then after the prop passes?